docs/load_balancer.md

```diff
@@ -0,0 +1,278 @@
+# Load balancing & publicly accessible services
+
+This document describes how kubernetes services are made accessible outside the cluster.
+
+There are two main mechanisms:
+
+* The service can be exposed through a load-balancer managed by k8s.
+* k8s can also expose services without an associated load-balancer; the service could be reached via DNS SRV records,
+or by a load-balancer configured by a helper process, or even by a client using the k8s API directly for discovery.
+_(new functionality)_
+
+## Public ports
+
+To be exposed outside the k8s cluster, the service must be on known ports on externally reachable
+IPs.  For version 1 public services, k8s listens on _every_ external IP address on every minion, and assigns
+a _single_ port cluster-wide for each port on public services.
+
+## Important scenarios
+
+The names in bold are the names we will use in the API.
```

Comment:
Naming things, how I love it.  There's power in the names...  Let's try some free-association, and see inf anything good emerges

* VIP, Node, Public
* Internal, External, Public
* IntraCluster, ExtraCluster, Public
* Cluster, Project, Public (very GCE specific)
* Cluster, Prod, Public

What would these mean in a VM-less installation?  If there were no nodes, does the middle mode just cease to work?

I dislike the "loadBalancer" moniker because it really talks about implementation and not general paradigm.